src/bootstrap/compile.rs

```diff
@@ -198,6 +198,10 @@ pub fn rustc<'a>(build: &'a Build, target: &str, compiler: &Compiler<'a>) {
     if !build.unstable_features {
         cargo.env("CFG_DISABLE_UNSTABLE_FEATURES", "1");
     }
+    // Flag that rust llvm is in use
+    if build.config.target_config.get(target).is_none() {
```

Comment:
The strange thing is that there seems to be no target returned at this point if --llvm-root has not been specified. If --llvm-root is specified then there is a target and it has an llvm_config set. I could make this code check for no target or a target with no llvm_config, but it seems like maybe there being no target is unexpected?

How do you usually debug this? For now I've just been printf debugging with println! or panic! calls.